### ISSUE TITLE
Build failure: buildRustPackage

### ISSUE THREAD

**Issue by greaka** [NixOS/nixpkgs#232052]
### Steps To Reproduce

1. build anything using rust build package with a git dependency that uses workspaces and `package.publish = true`.

Issue is here: https://github.com/NixOS/nixpkgs/blob/42bfe21cb089d04cfe5696eb14959113776de8e2/pkgs/build-support/rust/replace-workspace-values.py#L21

`table[key]` is not an iterable, because `publish = true` returns a bool.

### Build log

```
Traceback (most recent call last):
  File "/nix/store/ahm82yipa4px066krw1vqsrfzfg74l29-replace-workspace-values", line 104, in <module>
    main()
  File "/nix/store/ahm82yipa4px066krw1vqsrfzfg74l29-replace-workspace-values", line 84, in main
    changed |= replace_key(
  File "/nix/store/ahm82yipa4px066krw1vqsrfzfg74l29-replace-workspace-values", line 22, in replace_key
    if "workspace" in table[key] and table[key]["workspace"] is True:
TypeError: argument of type 'bool' is not iterable
```

### Notify maintainers

@winterqt @DeeUnderscore 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.13-arch1-1, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.0`
 - channels(root): `"nixpkgs"`
 - channels(greaka): `"nixpkgs-23.05pre438170.57dbff4228a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by DeeUnderscore** [comment]
#228928 addresses this

**Comment by greaka** [comment]
Thanks, I didn't find that PR while searching for it.

**Comment by DeeUnderscore** [comment]
Closing this as resolved by #228928. Feel free to reopen if this is still an issue with current master.
